php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1269 @@
+import org.sonar.plugins.php.api.tree.statement.StaticStatementTree;
+import org.sonar.plugins.php.api.tree.statement.SwitchCaseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.SwitchStatementTree;
+import org.sonar.plugins.php.api.tree.statement.ThrowStatementTree;
+import org.sonar.plugins.php.api.tree.statement.TraitAliasTree;
+import org.sonar.plugins.php.api.tree.statement.TraitMethodReferenceTree;
+import org.sonar.plugins.php.api.tree.statement.TraitPrecedenceTree;
+import org.sonar.plugins.php.api.tree.statement.TraitUseStatementTree;
+import org.sonar.plugins.php.api.tree.statement.TryStatementTree;
+import org.sonar.plugins.php.api.tree.statement.UnsetVariableStatementTree;
+import org.sonar.plugins.php.api.tree.statement.UseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.UseStatementTree;
+import org.sonar.plugins.php.api.tree.statement.WhileStatementTree;
+import org.sonar.plugins.php.api.tree.statement.YieldStatementTree;
+
+import static org.sonar.php.api.PHPKeyword.ABSTRACT;
+import static org.sonar.php.api.PHPKeyword.ARRAY;
+import static org.sonar.php.api.PHPKeyword.CLASS;
+import static org.sonar.php.api.PHPKeyword.EXTENDS;
+import static org.sonar.php.api.PHPKeyword.FINAL;
+import static org.sonar.php.api.PHPKeyword.IMPLEMENTS;
+import static org.sonar.php.api.PHPKeyword.INTERFACE;
+import static org.sonar.php.api.PHPKeyword.LIST;
+import static org.sonar.php.api.PHPKeyword.STATIC;
+import static org.sonar.php.api.PHPKeyword.TRAIT;
+import static org.sonar.php.api.PHPKeyword.USE;
+import static org.sonar.php.api.PHPKeyword.YIELD;
+import static org.sonar.php.api.PHPPunctuator.AMPERSAND;
+import static org.sonar.php.api.PHPPunctuator.ARROW;
+import static org.sonar.php.api.PHPPunctuator.COLON;
+import static org.sonar.php.api.PHPPunctuator.COMMA;
+import static org.sonar.php.api.PHPPunctuator.DOLLAR_LCURLY;
+import static org.sonar.php.api.PHPPunctuator.DOUBLEARROW;
+import static org.sonar.php.api.PHPPunctuator.DOUBLECOLON;
+import static org.sonar.php.api.PHPPunctuator.ELIPSIS;
+import static org.sonar.php.api.PHPPunctuator.EQU;
+import static org.sonar.php.api.PHPPunctuator.LBRACKET;
+import static org.sonar.php.api.PHPPunctuator.LCURLYBRACE;
+import static org.sonar.php.api.PHPPunctuator.LPARENTHESIS;
+import static org.sonar.php.api.PHPPunctuator.RBRACKET;
+import static org.sonar.php.api.PHPPunctuator.RCURLYBRACE;
+import static org.sonar.php.api.PHPPunctuator.RPARENTHESIS;
+
+public class NewPHPGrammar {
+
+  private final GrammarBuilder<InternalSyntaxToken> b;
+  private final TreeFactory f;
+
+  public NewPHPGrammar(GrammarBuilder<InternalSyntaxToken> b, TreeFactory f) {
+    this.b = b;
+    this.f = f;
+  }
+
+  public CompilationUnitTree COMPILATION_UNIT() {
+    return b.<CompilationUnitTree>nonterminal(PHPLexicalGrammar.COMPILATION_UNIT).is(
+      f.compilationUnit(b.optional(SCRIPT()), b.token(PHPLexicalGrammar.EOF))
+    );
+  }
+
+  public ScriptTree SCRIPT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)